src/OpenColorIO/ops/Lut3D/Lut3DOpData.h

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright (c) 2018 Autodesk Inc., et al.
+All Rights Reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are
+met:
+* Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+* Neither the name of Sony Pictures Imageworks nor the names of its
+  contributors may be used to endorse or promote products derived from
+  this software without specific prior written permission.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#ifndef INCLUDED_OCIO_LUT3DOPDATA_H
+#define INCLUDED_OCIO_LUT3DOPDATA_H
+
+#include <vector>
+
+#include <OpenColorIO/OpenColorIO.h>
+
+#include "Op.h"
+#include "ops/OpArray.h"
+
+OCIO_NAMESPACE_ENTER
+{
+class Lut3DOpData;
+typedef OCIO_SHARED_PTR<Lut3DOpData> Lut3DOpDataRcPtr;
+
+class Lut3DOpData : public OpData
+{
+public:
+
+    // Enumeration of the inverse 3D LUT styles.
+    enum InvStyle
+    {
+        INV_EXACT = 0,  // Exact, but slow, inverse processing
```

Comment:
If EXACT is the inverse of the forward tetrahedral interpolation, and FAST is the inverse of trilinear, could this be driven instead by the interpolation choice? I guess I'm wondering if having different semantics (and interface) for forward vs inverse LUT interpolation will be confusing... unless this is to point out the performance difference, as presumably EXACT would be slower than tetrahedral by a higher degree than FAST is slower than trilinear.

Reply:
EXACT or FAST can be used for CPU inversion of a forward tetrahedral or linear LUT.

Reply:
With both the forward interpolation style and the inverse style, when inverting twice, the original style is preserved.

Reply:
Responding with a related comment on the GPU implementation below.

Reply:
You raise an excellent point Michael.  Please take a look at Issue #621.  In summary, yes some more work on your point is needed but we would like to do it in a separate PR, so let's use Issue #621 to discuss what the best solution would be.

Reply:
Thanks for pointing out #621 . Somehow I missed that one.